packages/core/src/app/payment/paymentMethod/HostedPaymentMethod.tsx

```diff
@@ -168,6 +164,34 @@ class HostedPaymentMethod extends Component<
             selectedInstrument: find(instruments, { bigpayToken: id }),
         });
     };
+
+    private getPaymentOptions(): PaymentInitializeOptions {
+        const { onUnhandledError, method, language } = this.props;
+
+        const options = {
+            gatewayId: method.gateway,
+            methodId: method.id,
+        };
+
+        if (method.id === 'braintreepaypal') {
```

Comment:
I think we agreed to avoid adding further payment method specific logic to `core`.
Is it possbile to place this in its own repo?

Reply:
@bc-peng yes, it is possible. I have updated PR with an appropriate changes. Take a look pls